File: NavieraCruzApp/Modules/Finance/FinanceView.swift
import SwiftUI

struct FinanceView: View {
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                HStack {
                    Text("Dashboard Contable")
                        .font(Typography.title2())
                    Spacer()
                }
                .padding(.horizontal)
                
                // Resumen Ejecutivo
                VStack(spacing: 15) {
                    FinanceRow(title: "Facturación Mensual", amount: "$1.2M", isPositive: true)
                    FinanceRow(title: "Presupuesto Ejecutado", amount: "$850k", isPositive: nil)
                    FinanceRow(title: "Desviación Operativa", amount: "-$45k", isPositive: false)
                }
                .padding()
                .background(ColorTheme.background)
                .cornerRadius(15)
                .shadow(color: Color.black.opacity(0.05), radius: 5, x: 0, y: 2)
                .padding(.horizontal)
                
                // Desglose de Gastos Operativos por Buque
                VStack(alignment: .leading, spacing: 15) {
                    Text("Distribución de Gastos (Flota)")
                        .font(Typography.headline())
                        .foregroundColor(ColorTheme.textPrimary)
                        .padding(.bottom, 5)
                    
                    ExpenseItemRow(label: "Combustible y Lubricantes", amount: "$420,000", percentage: 0.50, color: .orange)
                    ExpenseItemRow(label: "Mantenimiento Técnico", amount: "$250,000", percentage: 0.30, color: .blue)
                    ExpenseItemRow(label: "Sueldos y Personal", amount: "$130,000", percentage: 0.15, color: .green)
                    ExpenseItemRow(label: "Tasas Portuarias y Seguros", amount: "$50,000", percentage: 0.05, color: .purple)
                }
                .padding()
                .background(ColorTheme.background)
                .cornerRadius(15)
                .shadow(color: Color.black.opacity(0.05), radius: 5, x: 0, y: 2)
                .padding(.horizontal)
            }
            .padding(.vertical)
        }
        .navigationTitle("Contable")
        .background(ColorTheme.background.ignoresSafeArea())
    }
}

struct FinanceRow: View {
    let title: String
    let amount: String
    let isPositive: Bool?
    
    var body: some View {
        HStack {
            Text(title)
                .font(Typography.body())
                .foregroundColor(ColorTheme.textSecondary)
            
            Spacer()
            
            Text(amount)
                .font(Typography.headline())
                .foregroundColor(
                    isPositive == nil ? ColorTheme.textPrimary :
                        (isPositive! ? ColorTheme.success : ColorTheme.danger)
                )
        }
        .padding(.vertical, 4)
    }
}

struct ExpenseItemRow: View {
    let label: String
    let amount: String
    let percentage: Double
    let color: Color
    
    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            HStack {
                Text(label)
                    .font(.caption)
                    .foregroundColor(ColorTheme.textSecondary)
                Spacer()
                Text(amount)
                    .font(.caption)
                    .fontWeight(.semibold)
                    .foregroundColor(ColorTheme.textPrimary)
            }
            
            GeometryReader { geometry in
                ZStack(alignment: .leading) {
                    RoundedRectangle(cornerRadius: 4)
                        .fill(ColorTheme.secondaryBackground)
                        .frame(height: 8)
                    
                    RoundedRectangle(cornerRadius: 4)
                        .fill(color)
                        .frame(width: geometry.size.width * CGFloat(percentage), height: 8)
                }
            }
            .frame(height: 8)
        }
        .padding(.vertical, 2)
    }
}

File: naviera_cruz_app/lib/modules/finance/finance_view.dart
import 'package:flutter/material.dart';
import '../../app/theme.dart';

class FinanceView extends StatelessWidget {
  const FinanceView({super.key});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);

    return Scaffold(
      appBar: AppBar(
        title: const Text("Contable", style: TextStyle(fontWeight: FontWeight.bold)),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              "Dashboard Contable",
              style: TypographyTheme.title2(context),
            ),
            const SizedBox(height: 16),
            
            // Executive Summary Card
            Card(
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(15)),
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Column(
                  children: [
                    _buildFinanceRow("Facturación Mensual", "\$1.2M", Colors.green),
                    const Divider(height: 20),
                    _buildFinanceRow("Presupuesto Ejecutado", "\$850k", theme.colorScheme.onSurface),
                    const Divider(height: 20),
                    _buildFinanceRow("Desviación Operativa", "-\$45k", ColorTheme.danger),
                  ],
                ),
              ),
            ),
            
            const SizedBox(height: 24),
            
            // Expenses breakdown Card
            Card(
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(15)),
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      "Distribución de Gastos (Flota)",
                      style: TypographyTheme.headline(context),
                    ),
                    const SizedBox(height: 20),
                    _buildExpenseRow("Combustible y Lubricantes", "\$420,000", 0.50, Colors.orange),
                    const SizedBox(height: 16),
                    _buildExpenseRow("Mantenimiento Técnico", "\$250,000", 0.30, Colors.blue),
                    const SizedBox(height: 16),
                    _buildExpenseRow("Sueldos y Personal", "\$130,000", 0.15, Colors.green),
                    const SizedBox(height: 16),
                    _buildExpenseRow("Tasas Portuarias y Seguros", "\$50,000", 0.05, Colors.purple),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildFinanceRow(String title, String amount, Color amountColor) {
    return Row(
      mainAxisAlignment: MainAxisAlignment.spaceBetween,
      children: [
        Text(title, style: const TextStyle(fontSize: 14, color: Colors.grey)),
        Text(
          amount,
          style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold, color: amountColor),
        ),
      ],
    );
  }

  Widget _buildExpenseRow(String label, String amount, double percentage, Color color) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Text(label, style: const TextStyle(fontSize: 12, color: Colors.grey)),
            Text(amount, style: const TextStyle(fontSize: 12, fontWeight: FontWeight.bold)),
          ],
        ),
        const SizedBox(height: 6),
        LayoutBuilder(
          builder: (context, constraints) {
            return Stack(
              children: [
                Container(
                  height: 8,
                  width: double.infinity,
                  decoration: BoxDecoration(
                    color: Colors.grey.withOpacity(0.15),
                    borderRadius: BorderRadius.circular(4),
                  ),
                ),
                Container(
                  height: 8,
                  width: constraints.maxWidth * percentage,
                  decoration: BoxDecoration(
                    color: color,
                    borderRadius: BorderRadius.circular(4),
                  ),
                ),
              ],
            );
          },
        ),
      ],
    );
  }
}
